core/src/actors/chain_manager/mod.rs

```diff
@@ -267,6 +299,29 @@ impl ChainManager {
         }
     }
 
+    fn keep_block_without_validation(&mut self, block: Block) -> Result<Hash, ChainManagerError> {
+        // Calculate the hash of the block
+        let hash: Hash = block.hash();
+
+        // Check if we already have a block with that hash
+        if let Some(_block) = self.blocks_to_validate.get(&hash) {
+            Err(ChainManagerError::BlockAlreadyExists)
```

Comment:
I think is unnecessary to err if the block already exist, the method could silently ignore this fact IMHO